### PR TITLE
fix: prebuild env

### PR DESCRIPTION
**Pull request by SwenSchaeferjohann**
- add env copy to prebuild step in webapp
- add docker auth to prevent docker hub image pull rate limits

### CODE REVIEW

File: web/webapp/copy-env.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env sh
+set -eux
+
+# copy .env.local.example to .env if .env doesn't already exist
+if [ ! -f .env ]; then
+    cp .env.local.example .env
+    echo ".env file created from .env.local.example"
+fi
+
+sleep 2
```
Comment (vadorovsky) [inline web/webapp/copy-env.sh:10]: Why is this sleep needed?

If you ware worried about fine not being actually copied yet, I would rather use `sync .env` (https://phoenixnap.com/kb/linux-sync#ftoc-heading-5 - this is forcing the kernel to actually write the file from cache to the disk)

File: web/webapp/package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "private": true,
   "scripts": {
     "dev": "next dev",
-    "prebuild": "chmod +x ./copy-circuits.sh && sh ./copy-circuits.sh",
+    "prebuild": "chmod +x ./copy-circuits.sh && sh ./copy-circuits.sh && chmod +x ./copy-env.sh && sh ./copy-env.sh",
```
Comment (vadorovsky) [inline web/webapp/package.json:7]: I didn't notice these `chmod`s before. You don't need them. Just do `chmod +x` locally and `git add` these files. Git is keeping track of executable bits of files.

**Review by vadorovsky — CHANGES_REQUESTED** [review]

File: web/webapp/copy-circuits.sh [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
 # copy build-circuits from zkjs into webapp/public
 SOURCE="./node_modules/@lightprotocol/zk.js/build-circuits/"
 DESTINATION="./public/"
 
-# mkdir -p "$DESTINATION"
 cp -LR "$SOURCE" "$DESTINATION"
 
 echo "Copied circuit files to $DESTINATION"
 
 sleep 5
```
Comment (sergeytimoshin) [inline web/webapp/copy-circuits.sh:14]: One more sleep, is it necessary?

**Review by vadorovsky — APPROVED** [review]
Awesome, thanks!

**Comment by SwenSchaeferjohann** [comment]
thx for the reviews and tips! 
i also just added docker auth to prevent rate limits on our docker pulls for the docker relayer test

**Pull request merged**